File: test/src/main/webapp/WEB-INF/cgi/jsonEncodeDecode.pl
#!D:/Perl64/bin/perl.exe
print "Content-type: text/html\n\n";

=pod
use JSON; 
my $text = '{"name":"John","age":38,"married":true}';
my $obj = decode_json ($text);
#my $obj = JSON->new->utf8->decode($text);
print "name = " . $obj->{"name"};
print ", age = " . $obj->{"age"};
print ", married = " . $obj->{"married"};


use JSON; 
my $text = '["John",38,true]';
my @arr = @{decode_json ($text)};
print "name = " . @arr[0] . ", age = " . @arr[1] . ", married = " . @arr[2];
=cut

=pod
package Person;
sub new {
   my $class = shift;	
   my $self = {
      name => shift,
      designation  => shift,
      contact  => shift,
   };	
   return $self;
}
use JSON; 
my $obj = new Person( "John", 38, true);
#my $text = JSON->new->utf8->encode($obj);
my $text = encode_json $obj;
print $text;
=cut

use JSON; 
my @arr = ( "John", 38, true);
#my $text = JSON->new->utf8->encode(\@arr);
my $text = encode_json (\@arr);
print $text;




=pod
use JSON;
my %rec_hash = ('a' => 1, 'b' => 2, 'c' => 3, 'd' => 4, 'e' => 5);
my $json = encode_json \%rec_hash;
print "$json\n";
=cut

File: test/src/main/webapp/WEB-INF/cgi/jsonEncodeDecode.py
#!D:/Python27/python.exe
import json
print "Content-Type: text/html\n"

'''
data = [ { 'a':'A', 'b':(2, 4), 'c':3.0 } ]
print 'DATA:', repr(data)

data_string = json.dumps(data)
print 'JSON:', data_string


obj = { 'name':'John', 'age':38, 'married':True }
text = json.dumps(obj, indent=2)
print text
#text = '{"age": 38, "married": true, "name": "John"}'

text = '{"name": "John", "age": 38, "married": true}'
obj = json.loads(text)
print 'name = ',obj['name'],', age = ',obj['age'],', married = ',obj['married']
'''
'''
encoder = json.JSONEncoder()
decoder = json.JSONDecoder()
obj = { 'name':'John', 'age':38, 'married':True }
text = encoder.encode(obj)
print text

obj = decoder.decode(text);
print 'name = ',obj['name'],', age = ',obj['age'],', married = ',obj['married']
	'''
	
	
import json


class MyObj(object):
    def __init__(self, s):
        self.s = s
    def __repr__(self):
        return '<MyObj(%s)>' % self.s
		
class Point(object):
    def __init__(self, x, y):
        self.x = x
        self.y = y
class User(object):
    def __init__(self, name, password):
        self.name = name
        self.password = password


class MyEncoder(json.JSONEncoder):    
    def default(self, obj):
		return obj.__dict__
		
def serialize(o):
    return o.__dict__


		
#obj = MyObj('internal data')
#obj = Point(10,20)
#print json.dumps(Point(10,20))
print json.dumps(Point(10, 20), default=serialize)
#obj = User('Alice A. Adams', 'secret')
#print obj
print MyEncoder().encode(Point(10, 20))

#print(json.dumps(obj, default=jdefault))